distributed/worker.py

```diff
@@ -2347,7 +2357,6 @@ def transition_ready_executing(
             )
 
         ts.state = "executing"
-        self._executing.add(ts)
```

Comment:
This line has moved to _ensure_computing. I'm not happy about it but I could not find a cleaner solution.